src/main/java/org/apache/sysds/runtime/instructions/cp/VariableCPInstruction.java

```diff
@@ -1060,6 +1054,36 @@ private void processWriteInstruction(ExecutionContext ec) {
 			HDFSTool.writeScalarToHDFS(ec.getScalarInput(getInput1()), fname);
 		}
 		else if( getInput1().getDataType() == DataType.MATRIX ) {
+			MatrixObject mo = ec.getMatrixObject(getInput1().getName());
+			int blen = Integer.parseInt(getInput4().getName());
+			LocalTaskQueue<IndexedMatrixValue> stream = mo.getStreamHandle();
+
+			if (stream != null) {
```

Comment:
Due to recompilation, this is visited the second. Since the stream is null the second time, it fails.

So, I tried setting the values explicitly. But, recompilation still happening.

```java
mo.updateDataCharacteristics(mc);
HDFSTool.copyFileOnHDFS(fname, mo.getFileName());
mo.setDirty(false);
```

Reply:
I would recommend to simple probe if a stream exists and not consume the matrix if it does.

Reply:
I've not understood this part.